src-ext/TypeScript/ggvega/src/encodingNameByGeom.ts

```diff
@@ -0,0 +1,76 @@
+import * as GG from '../../ggschema/src/index';
+import {contains} from './utils';
+
+/**
+ * Get encoding name from aesthetic name
+ *
+ * @remarks
+ * The mapping of a ggplot aesthetic name to an encoding name depends
+ * on the Geom under consideration.
+ *
+ * **Called by**
+ * @see layerByItmLayer
+ *
+ * @param aesName - `GGEncodingKey` ggplot aesthetic-name
+ * @param ggGeom - `GG.GeomSet` ggschema GeomSet object
+ *
+ * @returns
+ */
+export function encodingNameByGeom(aesName: GGEncodingKey, ggGeomSet: GG.GeomSet): VLEncodingKey {
+  // keys: names of ggplot2 aesthetics
+  // values: names of Vega-Lite encodings
+  //NOTE @wenyu: Add `stroke`, What's `weight`
+  const encodingMap = encodingMapDefault;
+
+  // validate
+  if (!contains(Object.keys(encodingMap), aesName)) {
+    throw new Error('ggplot object contains unsupported aesthetic: ' + aesName);
+  }
+
+  // translate
+
+  // exceptions
+  if (ggGeomSet.geom.class == 'GeomLine') {
+    encodingMap.size = 'strokeWidth';
+  }
+
+  if (ggGeomSet.geom.class == 'GeomBar') {
+    encodingMap.size = 'strokeWidth';
+  }
+
+  return encodingMap[aesName] as VLEncodingKey;
+}
+
+//NOTE @wenyu: keyof maybe a better way than enum. But it doesn't work for Vl.Encoding, because VL.Encoding has too many keys.
```

Comment:
Could we move this material (everything from here to the end of the file) to the start of the file? If we have to add an encoding definition, this will make it easier to find.

Reply:
done.